InsiteTeamTask/Controllers/AttendanceController.cs

```diff
@@ -2,22 +2,43 @@
 using InsiteTeamTask.Models;
 using InsiteTeamTask.Repositories;
 using Microsoft.AspNetCore.Mvc;
+using System.Linq;
 
 namespace InsiteTeamTask.Controllers
 {
-    [Route("api/[controller]")]
-    [ApiController]
-    public class AttendanceController : ControllerBase
-    {
-        // GET api/values
-        [HttpGet]
-        public ActionResult<IEnumerable<Attendance>> Get()
-        {
-            var repo = new DataRepository();
+	[Route("api/[controller]")]
+	[ApiController]
+	public class AttendanceController : ControllerBase
+	{
+		private IDataRepository _dataRepository;
 
-            var attendance = repo.GetAttendanceListFor(gameNumber: 3);
+		public AttendanceController(IDataRepository dataRepository)
+		{
+			this._dataRepository = dataRepository;
+		}
 
-            return Ok(attendance);
-        }
-    }
+		// GET api/values
+		[HttpGet]
+		public ActionResult<IEnumerable<Attendance>> Get(string productId, int? seasonId, int? gameId)
```

Comment:
This probably should be split into two separate controller actions:
`public ActionResult<IEnumerable<Attendance>> GetByProduct(string productId)`
and
`public ActionResult<IEnumerable<Attendance>> GetByGame(int seasonId, int gameId)`

Also, since the GetByGame method expects a seasonId and a gameId, neither of these should be of type `int?`. The controller will work out which action is being requested by analysing the parameters.